flink-streaming-java/src/test/java/org/apache/flink/streaming/util/MockStreamTask.java

```diff
@@ -104,7 +104,11 @@ public String getName() {
 		return name;
 	}
 
-	@Override
+	/**
+	 * Checkpoint lock in {@link StreamTask} is replaced by {@link StreamTaskActionExecutor}.
+	 * <code>getCheckpointLock</code> method was moved from to the {@link org.apache.flink.streaming.runtime.tasks.SourceStreamTask SourceStreamTask}.
+	 */
+	@Deprecated
```

Comment:
Do we need this method? It's a bit strange that it was removed from the production class, but mock still needs it.

Reply:
Yes, a lot of tests are using checkpoint lock, so I decided to keep it. The tests I think should be changed incrementally.